pkgs/applications/emulators/mame/default.nix

```diff
@@ -109,17 +105,15 @@ stdenv.mkDerivation rec {
   ];
 
   patches = [
-    # by default MAME assumes that paths with stock resources
-    # are relative and that you run MAME changing to
-    # install directory, so we add absolute paths here
-    ./emuopts.patch
+    # by default MAME assumes that paths with stock resources are relative and
+    # that you run MAME changing to install directory, so we add absolute paths
+    # here
+    (substituteAll {
+      mamePath = "$out/opt/mame";
+      src = ./001-use-absolute-paths.diff;
+    })
```

Comment:
This doesn't work. This is part of the result file:

```
        { OPTION_HASHPATH ";hash_directory;hash",            "hash;$out/opt/mame/hash",      core_options::option_type::STRING,     "path to software definition files" },
        { OPTION_SAMPLEPATH ";sp",                           "samples;$out/opt/mame/samples",   core_options::option_type::STRING,     "path to audio sample sets" },
        { OPTION_ARTPATH,                                    "artwork;$out/opt/mame/artwork",   core_options::option_type::STRING,     "path to artwork files" },
```

Also, `placeholder "out"` doesn't work in this case because of issue https://github.com/NixOS/nixpkgs/issues/135438.

Reply:
This is why we use `substituteInPlace --subst-var-by` later on.

Reply:
I will "document" this.